Further errors-related adjustments (#4492)
statelyai/xstate: .changeset/olive-months-shave.md

---
'xstate': major
---

> All errors caught while executing the actor should now consistently error its snapshot and should be reported to the outermost `error` listener if they stay unhandled.

Reviewer: ```suggestion
All errors caught while executing the actor should now consistently include the error in its `snapshot.error` and should be reported to the closest `error` listener.
```
